src/block-ref.ts

```diff
@@ -68,12 +63,15 @@ export function createBlockRefMiddleware({
 
     // perform child request
     log('Performing another request %o', childRequest);
-    const childRes: PendingJsonRpcResponse<Block> = await pify(
-      provider.sendAsync,
-    ).call(provider, childRequest);
     // copy child response onto original response
-    res.result = childRes.result;
-    res.error = childRes.error;
```

Comment:
hi @legobeat the EIP-1193 changes require the usage of `request` method which return a promise of the result on success or throws an error in other cases  instead of `sendAsync` which takes a callback that can return both error and result

Reply:
Correct. The `request` method now guarantees that `response.error` will be thrown: https://github.com/MetaMask/core/blob/e441444f473ffe75257000f0948d1e4590790224/packages/eth-json-rpc-provider/src/safe-event-emitter-provider.ts#L94-L102